contracts/service-registry/src/contract/execute.rs

```diff
@@ -34,10 +37,35 @@ pub fn register_service(
             description,
         },
     )?;
-
     Ok(Response::new())
 }
 
+fn ensure_authorization_max_limit_respected(
+    deps: &DepsMut,
+    service_name: &String,
+    verifiers: &[Addr],
+) -> Result<(), ContractError> {
+    let max_limit = state::service(deps.storage, service_name, None)?.max_num_verifiers;
+    if let Some(max_limit) = max_limit {
+        let authorzied_verifier_count =
+            state::number_of_authorized_verifiers(deps.storage, service_name)?;
+
+        let additional_authorizations =
+            state::count_verifiers_becoming_authorized(deps.storage, service_name, verifiers)?;
+
+        let total_after_update = authorzied_verifier_count
+            .checked_add(additional_authorizations)
+            .ok_or(ContractError::AuthorizedVerifiersExceedu16)?;
+
+        ensure!(
+            total_after_update <= max_limit,
+            ContractError::VerifierLimitExceed
+        );
+    }
+
+    Ok(())
+}
+
 pub fn update_verifier_authorization_status(
```

Comment:
suggestion for a slightly modified approach:
- let `state` fully control how the verifier state is modelled, i.e. `execute` gives the command to change the auth state, but doesn't care how that is done
- after the authorization call get the new number
- ensure the new number is below or equal to the limit

you don't have to implement a "simulation mode", on error everything gets rolled back